torch/_library/custom_ops.py

```diff
@@ -660,12 +661,18 @@ def adinplaceorview_impl(keyset, *args, **kwargs):
                 # Handle view + mutation that are in the schema
                 return original_kernel.call_boxed(keyset, *args, **kwargs)
 
-            lib.impl(
-                self._name,
-                adinplaceorview_impl,
-                "ADInplaceOrView",
-                with_keyset=True,
-            )
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    "ignore",
+                    message="Warning only once for all operators",
+                    category=UserWarning,
+                )
+                lib.impl(
+                    self._name,
+                    adinplaceorview_impl,
+                    "ADInplaceOrView",
+                    with_keyset=True,
+                )
```

Comment:
@albanD I think the problem is that there's something registered at line 644. Can we just not register that if we're going to registering something here?

Reply:
Also can you add a test that torch.library.custom_op with or without mutates_args raises no warnings?

Reply:
It's a bit tricky as they do two different things this one does inplace bump based on the user-provided inplace) and the other is in C++ (view tracking + schema refcount bump). Also one is in python and one in C++ so hard to have them be one function.

Reply:
I'm confused, the code currently looks like:
```py
if schema._is_view_op() or schema.is_mutable:
    # this calls m.impl
    lib.m.register_ad_inplace_or_view_fallback(self._name) 
    ...
if schema.is_mutable:    
     ...
     lib.impl(self._name, adinplaceorview_impl, "ADInplaceOrView", with_keyset=True,)
```
Can we rewrite this into:
```py
if schema._is_view_op():
    # this calls m.impl
    lib.m.register_ad_inplace_or_view_fallback(self._name) 
    ...
elif schema.is_mutable:    
     ...
     lib.impl(self._name, adinplaceorview_impl, "ADInplaceOrView", with_keyset=True,)
```
? Operators are either view ops or mutable